File: 167. Two Sum II - Input Array Is Sorted/python/two_sum_ii_input_array_is_sorted.py
from typing import List


class Solution:
    def twoSumB(self, numbers: List[int], target: int) -> List[int]:
        f = 0
        l = len(numbers) - 1
        while l > f:
            s = numbers[f] + numbers[l]
            if s == target:
                return f+1, l+1
            if s > target:
                l -= 1
            if s < target:
                f += 1
    
    def twoSumD(self, numbers: List[int], target: int) -> List[int]:
        d = {}
        for i,v in enumerate(numbers):
            r = d.get(target - v, None)
            if not r is None:
                return r + 1, i + 1
            d[v] = i
        

File: 167. Two Sum II - Input Array Is Sorted/go/two_sum_ii_input_array_is_sorted.go
package two_sum_ii_input_array_is_sorted

func twoSum(numbers []int, target int) []int {
    f := 0
    l := len(numbers) - 1
    for l > f {
        s := numbers[f] + numbers[l]
        if s == target {
            return []int{f+1, l+1}
        }
        if s < target {
            f += 1
        }
        if s > target {
            l -= 1
        }
    }
    return []int{}
}